### PR TITLE
initial goma support

### PR DESCRIPTION
**Pull request by mihaiplesa**
Based on https://github.com/brave/brave-core/pull/8202 
For https://github.com/brave/devops/issues/4615

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by keur — APPROVED** [review]
LGTM. Built both a component and non-component build with goma. Works great.

**Review by antonok-edm — APPROVED** [review]

File: build/commands/lib/util.js [hunk 1]
```diff
@@ -513,6 +518,17 @@ const util = {
...
     ]
+
+    if (config.useGoma()) {
+      const gomaLoginInfo = util.runProcess('goma_auth', ['info'], options)
+      if (gomaLoginInfo.status !== 0) {
+        console.log('Login required for using Goma. This is only needed once')
+        util.run('goma_auth', ['login'], options)
+      }
+      util.run('goma_ctl', ['ensure_start'], options)
+      ninjaOpts.push('-j', (os.cpus().length + 2) * 10)
```
Comment (keur) [inline build/commands/lib/util.js:529]: Can we log this CPU count?

File: build/commands/lib/config.js [hunk 2]
```diff
@@ -502,6 +505,12 @@ Config.prototype.update = function (options) {
     this.is_asan = false
   }
 
+  if (options.use_goma) {
+    this.use_goma = true
+  } else {
+    this.use_goma = false
+  }
```
Comment (keur) [inline build/commands/lib/config.js:512]: nit: `this.use_goma = options.use_goma`

**Review by keur — APPROVED** [review]
LGTM. Been testing this a lot, works great.

**Pull request merged**